### ISSUE TITLE
Debian Stretch Apache 5.6 Dockerfile leaves unnecessary build-time dependencies in layer, bloating image

### ISSUE THREAD

**Issue by wyqydsyq** [docker-library/php#778]
The [Dockerfile](https://github.com/docker-library/php/blob/783878384a8f3953ed571e5a34ba0fe546726c85/5.6/stretch/apache/Dockerfile#L32) installs a bunch of build-time dependencies, apparently for "phpize" and never removes them, this results in a `209MB` layer:
```
<missing>           2 weeks ago         /bin/sh -c #(nop)  ENV PHP_INI_DIR=/usr/loca…   0B
<missing>           2 weeks ago         /bin/sh -c apt-get update && apt-get install…   209MB
<missing>           2 weeks ago         /bin/sh -c #(nop)  ENV PHPIZE_DEPS=autoconf …   0B
```

Which arguably provides little to no benefit to most users of this image, a lot of people don't use PECL extensions let alone phpize to install them and in our case, we're explicitly installing them with the PECL installer in our own image.

In the odd case someone does want to use phpize, there should be an alternative image with it enabled, instead of pointlessly bloating the base apache image.

For some reason, Docker Hub misreports the image as being `130MB`:
```
5.6-apache 130 MB
Last update: 2 days ago

```

But in reality upon pulling it, you find it's actually a whopping `355MB`:
```
php    5.6-apache          a62c5acfd6c5        2 days ago          355MB
```

I'm guessing a lot of this bloat is due to the completely unnecessary `209MB` layer mentioned.

**Comment by wyqydsyq** [comment]
In contrast, the Alpine-based image prevents these dependencies remaining by wrapping them all into a virtual package called `.build-deps` which gets deleted:
https://github.com/docker-library/php/blob/783878384a8f3953ed571e5a34ba0fe546726c85/5.6/alpine3.8/cli/Dockerfile#L159

Distributions like Debian and Ubuntu using `apt-get` should be able to support an equivalent functionality if including phpize in the base image is absolutely necessary (it seems completely redundant to me though): https://www.unixdaemon.net/linux/creating-debian-virtual-packages/

**Comment by yosifkit** [comment]
Duplicate of #769  (see also #513 #438 #557 #751 #716)

Docker Hub reporting 130MB is the compressed  size. which is the amount downloaded from the Hub, while 355MB is the on-disk extracted size.

----------------

On a side note: 5.6 is end of life (https://secure.php.net/supported-versions.php, #768) and was only added back for a security fix (#775) and will likely be removed again soon.  The images will still be available to pull, but they will no longer be rebuilt if the base image get updated.
